src/main/java/cs/f10/t1/nursetraverse/model/HistoryManager.java

```diff
@@ -39,46 +40,50 @@ public HistoryManager(int maxSize) {
-     * {@code command} was executed.
+     * Records the specified command and state in the history. The {@code patientBook} and {@code appointmentBook}
+     * represents the state before the {@code command} was executed.
      *
      * @param command the command that caused the change in state
      * @param patientBook the state before the command was executed
+     * @param appointmentBook the state before the command was executed
      */
-    public void pushRecord(MutatorCommand command, PatientBook patientBook) {
+    public void pushRecord(MutatorCommand command, PatientBook patientBook, AppointmentBook appointmentBook) {
         if (command instanceof UndoCommand || command instanceof RedoCommand) {
             return;
         }
-        pushToHistory(command, patientBook);
+        pushToHistory(command, patientBook, appointmentBook);
         redoStack.clear(); // Invalidate redo stack to avoid branching
     }
 
     /**
      * Records the specified command and state in the history. This is different from
-     * {@link #pushRecord(MutatorCommand, PatientBook)} in that the {@code redoStack} is not cleared to facilitate
-     * pushing {@link #popRedo(PatientBook)}ed records back into the history while maintaining the redo stack.
+     * {@link #pushRecord(MutatorCommand, PatientBook, AppointmentBook)} in that the {@code redoStack} is not cleared to
+     * facilitate pushing {@link #popRedo(PatientBook, AppointmentBook)}ed records back into the history while
+     * maintaining the redo stack.
      */
-    private void pushToHistory(MutatorCommand command, PatientBook patientBook) {
+    private void pushToHistory(MutatorCommand command, PatientBook patientBook, AppointmentBook appointmentBook) {
         if (history.size() == maxSize) {
             history.remove(0);
         }
-        history.add(new HistoryRecord(command, patientBook));
+        history.add(new HistoryRecord(command, patientBook, appointmentBook));
     }
 
     /**
      * Removes the most recent record from the history and returns it as an {@code Optional}, or an empty
      * {@code Optional} if there are no records in the history. Records the undo with the specified
-     * {@code} currentPatientBook so it can be undone.
+     * {@code currentPatientBook} and {@code currentAppointmentBook} so it can be undone.
      *
      * @param currentPatientBook the current state of the {@code PatientBook} model, stored for redo purposes
+     * @param currentAppointmentBook the current state of the {@code AppointmentBook} model, stored for redo purposes
      * @return an {@code Optional} describing the most recent record in the history, or an empty {@code Optional} if
      *         the history is empty
      */
-    public Optional<HistoryRecord> popRecord(PatientBook currentPatientBook) {
+    public Optional<HistoryRecord> popRecord(PatientBook currentPatientBook, AppointmentBook currentAppointmentBook) {
```

Comment:
These look more like a new feature rather than a bug fix

Reply:
Not really bug fix, just adding appointment book to history manager so that undo and redo works on that.